File: src/inventory/routes/inventoryRoutes.js
const express = require('express');
const router = express.Router();
const InventoryItem = require('../models/InventoryItem'); // Adjust the path as necessary

// POST endpoint to add new goods
router.post('/', async (req, res) => {
  const { name, category, pricePerItem, description, countInStock } = req.body;
  try {
    const newItem = new InventoryItem({
      name,
      category,
      pricePerItem,
      description,
      countInStock
    });
    await newItem.save();
    res.status(201).send(newItem);
  } catch (error) {
    res.status(400).send(error);
  }
});

// POST endpoint to deduct goods
router.post('/deduct', async (req, res) => {
  const { itemId, quantity } = req.body;
  try {
    const item = await InventoryItem.findById(itemId);
    if (item && item.countInStock >= quantity) {
      item.countInStock -= quantity;
      await item.save();
      res.status(200).send(item);
    } else {
      res.status(400).send({ message: 'Insufficient stock or item not found' });
    }
  } catch (error) {
    res.status(500).send(error);
  }
});

// PUT endpoint to update goods
router.put('/:itemId', async (req, res) => {
  const { name, category, pricePerItem, description, countInStock } = req.body;
  const { itemId } = req.params;
  try {
    const item = await InventoryItem.findByIdAndUpdate(itemId, {
      name,
      category,
      pricePerItem,
      description,
      countInStock
    }, { new: true });
    if (item) {
      res.status(200).send(item);
    } else {
      res.status(404).send({ message: 'Item not found' });
    }
  } catch (error) {
    res.status(400).send(error);
  }
});

module.exports = router;

File: src/inventory/routes/inventoryRoutes.py
from flask import Blueprint, request, jsonify
from db import db
from inventoryitems import InventoryItem

inventory_bp = Blueprint('inventory', __name__, url_prefix='/inventory')

@inventory_bp.route('/', methods=['POST'])
def add_goods():
    data = request.get_json()
    new_item = InventoryItem(name=data['name'], category=data['category'],
                             price_per_item=data['pricePerItem'], description=data['description'],
                             count_in_stock=data['countInStock'])
    db.session.add(new_item)
    db.session.commit()
    return jsonify(new_item), 201

@inventory_bp.route('/deduct', methods=['POST'])
def deduct_goods():
    data = request.get_json()
    item = InventoryItem.query.get(data['itemId'])
    if item and item.count_in_stock >= data['quantity']:
        item.count_in_stock -= data['quantity']
        db.session.commit()
        return jsonify(item), 200
    else:
        return jsonify({'error': 'Not enough stock or item not found'}), 400

@inventory_bp.route('/<int:item_id>', methods=['PUT'])
def update_goods(item_id):
    data = request.get_json()
    item = InventoryItem.query.get(item_id)
    if item:
        item.name = data.get('name', item.name)
        item.category = data.get('category', item.category)
        item.price_per_item = data.get('pricePerItem', item.price_per_item)
        item.description = data.get('description', item.description)
        item.count_in_stock = data.get('countInStock', item.count_in_stock)
        db.session.commit()
        return jsonify(item), 200
    else:
        return jsonify({'error': 'Item not found'}), 404
